### PR TITLE
feat(gwyneth): incorporate booster sample contracts

### PR DESCRIPTION
**Pull request by adaki2004**
There are some questions ariesed, i tried to asked them in form of some comments.

Also some libraries/contracts/interfaces like: 
`IExternalL2`, `IExternalL2Bridge`, `DemoReadFromExternalL2` , `Create2`, `IProxy`, `WalletProxy`, `SmartWallet`, `WalletFactory` are missing because first wanted to try with a simple erc20.

### CODE REVIEW

File: packages/protocol/contracts/gwyneth/XChain.sol [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+    struct AsyncBusProofV2 {
+        uint blockNumber;
+        uint busID;
+    }
+
+    // Only stored on L1
+    // Currently getBlockHash() is not supported via the new Taiko Gwyneth
+    //ITaiko public taiko;
+    // todo (@Brecht): XChain has a bus property but Bus is an XChain (inherits). It does not make too much sense to me, or maybe i'm missing the point ?
```
Comment (Brechtpd) [inline packages/protocol/contracts/gwyneth/XChain.sol:27]: The bus might still be needed for async transactions to pass data from one chain to the other, but I guess to keep things simple let's assume synchronous everywhere for now.

File: packages/protocol/contracts/gwyneth/XChain.sol [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                revert();
+            }
+        }
+        _;
+    }
+
+    // These could also be exposed using a precompile because we could get them from public input, 
+    // but that requires extra work so let's just fetch them from L1 for now
+    function getBlockHash(uint chainID, uint blockID) external view xExecuteOn(EVM.l1ChainId) returns (bytes32) {
+         // todo(@Brecht): Currently not supported or well, at least TaikoL1 does not have it with the current design.
```
Comment (Brechtpd) [inline packages/protocol/contracts/gwyneth/XChain.sol:99]: Right, outdated! Hmmmm only needed for async case so let's skip for now.

**Comment by Brechtpd** [comment]
Let's assume a wondrous fully synchronous world :rainbow: . XCALLOPTIONS is able to do any call to any chain, so we can avoid the Bus for now.

**Comment by adaki2004** [comment]
> Let's assume a wondrous fully synchronous world 🌈 . XCALLOPTIONS is able to do any call to any chain, so we can avoid the Bus for now.

Currently i can move synch-only code (from `Bus` to `XChain`), like `write()` + `consume()` (async case) but the `xFunction` modifier is in the `xChain` by default, and it handles async path too (`EVM.chainId() == fromChainId`).

Questions:
So basically you saying now, if a message is sync, xTransfer is never called on sourceChain ? If so, i think the balance calculation is not correct in XChainToken. (i guess we need to update balance on both for example).

Would be cool to go thru diff. `xTransfer` scenarios:
- L1 -> L2x
- L2x -> L2y
(Would be cool to cover asycn + sync scenarios, but maybe sync first, if we assume a synch world)

**Pull request merged**